korgw-sample/src/commonMain/kotlin/com/soywiz/korgw/sample/KorgwSample.kt

```diff
@@ -68,144 +53,103 @@ class KCube {
         name = "MY_PROG"
     )
 
-    fun run() = DefaultGameWindow.mainLoop {
-            configure(600, 600, "Multicolored cube", fullscreen = false)
+    var projMat = Matrix3D().setToPerspective(aov, ar, near, far)
+    var viewMat = Matrix3D().setToTranslation(xPos, yPos, zPos)
+    var modlMat = Matrix3D().identity()
+
+    fun run() = Korio {
+        DefaultGameWindow.loop {
+            //TODO: why can't I use width and height vars here ?
+            configure(400, 400, "Multicolored cube", fullscreen = false)
+            addEventListener<ReshapeEvent> {
+                reshape(it, ag)
+            }
             addEventListener<RenderEvent> {
                 render(ag)
             }
         }
-
-    fun init(ag:AG) {
-        //val gl = (ag as AGAwt).gl ??
-        //gl.clearColor(0f,0f,1f,1f)
-        //gl.clearDepthf(1.0f)
-        //gl.enable(gl.DEPTH_TEST)
-        // gl.depthFunc(gl.LEQUAL)
     }
 
-    private var vertices: AG.Buffer? = null
-    private var verticesData: FloatArrayList? = null
+    fun reshape(event: ReshapeEvent, ag: AG) {
+        width = event.width
+        height = event.height
+        if (height <= 0)
+            height = 1
+        ar = (width.toFloat() / height.toFloat())
+        ag.setViewport(0, 0, width, height)
+        projMat = Matrix3D().setToPerspective(aov, ar, near, far)
+    }
 
-    fun render(ag:AG) {
-        if (!init) {
-            init(ag)
-            init = true
+    fun render(ag: AG) {
+        ag.clear(color = Colors.BLACK, depth = far)
+
+        modlMat = Matrix3D().identity().setToRotation(rquad, rotAxis);
+
+        val points = mutableListOf<Float>()
```

Comment:
kds provides a `FloatArrayList` that is far more efficient and works the same. (Check the other `verticesData = FloatArrayList()`).
As a sample works fine, but when using large sets `MutableList` uses java's `ArrayList` that does boxing all the way so creates 6 objects for each vertex you have.